### PR TITLE
Add auto source sync to the first file process.

### PR DESCRIPTION
**Pull request by knightburton**
IoT.jsCode-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

### CODE REVIEW

**Comment by knightburton** [comment]
@daeyeon This is a solution for the first file sync, but the auto sync will load the iotjs core file(s) if the users are built the iotjs with `--no-snapshot` option. Otherwise the auto sync will load only the users code.

**Comment by rtakacs** [comment]
@knightburton If you know the names of the IoT.js core files then you can prevent to load these files into the editor.

File: src/app/session.js [hunk 1]
```diff
@@ -103,6 +103,7 @@ export default class Session {
     this._jerrySource = {
       name: '',
       source: '',
+      autoSourceSync: true,
```
Comment (daeyeon) [inline src/app/session.js]: Can we call `reset` here? Initializing properties is scattered.

Comment (knightburton) [reply src/app/session.js]: Yes, you are right.

**Comment by knightburton** [comment]
@rtakacs There are a few core files in the IoT.js, like `module.js`, `iotjs.js` and every builtin module. How we should decide that which will be prevented or not, and this would be an other list that we should to maintain. However, I am open minded for this.

File: src/app/session.js (hunk 1 above)
Comment (daeyeon) [reply src/app/session.js]: Oh. My query was whether we can call `this.reset()`, here, in the `constructor`. I guess that we could move the code here into `reset()` somehow.

**Comment by daeyeon** [comment]
@knightburton Oh. My query was whether we can call `this.reset()` in the constructor instead of initializing properties one by one. I guess that we could move the code here into `reset()` somehow.

**Comment by knightburton** [comment]
@daeyeon Oh, I see now. Do you want to init everything in a single function call and that would be called in the constructor and in the "reset" section after connection close, am I right?

**Comment by daeyeon** [comment]
@knightburton exactly!

**Comment by knightburton** [comment]
@daeyeon All right, is that okay if I leave this PR here and create a new one for that?

**Comment by yichoi** [comment]
LGTM

**Pull request merged**